src/main/java/com/hms_networks/sc/cumulocity/data/CConnectorDataMgr.java

```diff
@@ -98,6 +99,9 @@ public static void checkForHistoricalDataAndSend(CConnectorMqttMgr mqttMgr) {
       } else if (mqttMgr == null) {
         Logger.LOG_WARN(
             "The MQTT manager is not available to send historical data. Skipping data poll!");
+      } else if (mqttMgr.getLastKnownMqttStatusCode() != MqttStatusCode.CONNECTED) {
```

Comment:
This could be an excessive amount of logging.

Reply:
This is consistent with the other related logging, including lines 99-101 (above) and the low memory warning. 

Do you think the rest of the related logging should be updated to reduce excessive output as well?

Reply:
I think the memory issue should resolve after garbage connection, but the connection could be down for hours. I'm never a fan of excessive logging. I would prefer log on change. If you want to add remove excessive connection warnings to backlog, that is fine.

Reply:
I will add an issue to address excessive logging. This may also be resolved as a side effect of migrating to the abstract connector framework soon, but nonetheless will be noted and resolved/tested.